File: Algorithms/Easy/two_sum.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        d = {}

        for idx, num in enumerate(nums):
            k = target - num
            if k in d.keys():
                 return [d[k], idx]
            d[num] = idx

File: Algorithms/Easy/two_sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> d;
        for (int i = 0; i != nums.size(); ++i) {
            int k = target - nums[i];
            if (d.contains(k)) {
                return {d[k], i};
                break;
            }
            d[nums[i]] = i;
        }
        return {0, 0};
    }
};
